Bottom: 8124969d1d4f1877a120c587f4fbafb4ab74f59c
Top:    7579efc7137120b2a055e576d8796e2ed0388c6c
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-26 22:04:30 +0000

Fix usage of hash tables - don't free the inserted key!


---

diff --git a/src/hid/step/step_writer.c b/src/hid/step/step_writer.c
index 89bbfd4..784145a 100644
--- a/src/hid/step/step_writer.c
+++ b/src/hid/step/step_writer.c
@@ -209,7 +209,6 @@ step_cartesian_point (step_file *file, char *name, double x, double y, double z)
 
   g_hash_table_insert (file->cartesian_point_hash, line, GINT_TO_POINTER (file->next_id));
   fprintf (file->f, "#%i=CARTESIAN_POINT(%s);\n", file->next_id, line);
-  g_free (line);
 
   return file->next_id++;
 }
@@ -230,7 +229,6 @@ step_direction (step_file *file, char *name, double x, double y, double z)
 
   g_hash_table_insert (file->direction_hash, line, GINT_TO_POINTER (file->next_id));
   fprintf (file->f, "#%i=DIRECTION(%s);\n", file->next_id, line);
-  g_free (line);
 
   return file->next_id++;
 }
@@ -283,7 +281,6 @@ step_vector (step_file *file, char *name, step_id orientation, double magnitude)
 
   g_hash_table_insert (file->vector_hash, line, GINT_TO_POINTER (file->next_id));
   fprintf (file->f, "#%i=VECTOR(%s);\n", file->next_id, line);
-  g_free (line);
 
   return file->next_id++;
 }
